### DISCUSSION TITLE
How to load dynamic component like Laravel components ?

### DISCUSSION THREAD

**Discussion by kossa** [livewire/livewire#2424] [1. Help]
Hello,

I have custom fields(text, textarea, select, file ....) I dont want to fill my page with multiple if or switch, need to load component like Laravel here : https://laravel.com/docs/8.x/blade#dynamic-components

```
<x-dynamic-component :component="$componentName" class="mt-4" />
```

**Comment by peterfox** [comment]
You can use the `@livewire($componentName, ['class' => 'mt-4'])` syntax. But I don't know if it can work with the html style system annoyingly.

**Reply by MACscr** [reply]
Mind expanding how one would set that component name by default and then with livewire events?

**Comment by aurelien31** [comment]
There is also the new way of doing, using the latest version with `<livewire:is :component="$componentName">`. 

See PR https://github.com/livewire/livewire/pull/2502 ...
